api/v1alpha1/externaldnsconfiguration_types.go

```diff
@@ -0,0 +1,131 @@
+	// +kubebuilder:validation:MaxItems:=20
+	// +kubebuilder:validation:items:Pattern:=`(?i)\/subscriptions\/(.{36})\/resourcegroups\/(.+?)\/providers\/Microsoft.network\/(dnszones|privatednszones)\/(.+)`
+	// +listType:=set
+	DNSZoneResourceIDs []string `json:"dnsZoneResourceIDs"`
+
+	// ResourceTypes is a list of Kubernetes resource types that the ExternalDNS controller should manage. The supported resource types are 'ingress' and 'gateway'.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems:=1
+	// +kubebuilder:validation:items:Enum:=ingress;gateway
+	// +listType:=set
+	ResourceTypes []string `json:"resourceTypes"`
+
+	// Identity contains information about the identity that ExternalDNS will use to interface with Azure resources.
+	// +kubebuilder:validation:Required
+	Identity ExternalDNSConfigurationIdentity `json:"identity"`
+
+	// Filters contains optional filters that the ExternalDNS controller should use to determine which resources to manage.
+	// +optional
+	Filters ExternalDNSConfigurationFilters `json:"filters,omitempty"`
+}
+
+// ExternalDNSConfigurationIdentity contains information about the identity that ExternalDNS will use to interface with Azure resources.
+type ExternalDNSConfigurationIdentity struct {
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^[a-z0-9][-a-z0-9\.]*[a-z0-9]$`
+
+	// ServiceAccount is the name of the Kubernetes ServiceAccount that ExternalDNS will use to interface with Azure resources. It must be in the same namespace as the ExternalDNSConfiguration.
+	ServiceAccount string `json:"serviceAccount"`
+}
+
+type ExternalDNSConfigurationFilters struct {
+	// GatewayLabels contains key-value pairs that the ExternalDNS controller will use to filter the Gateways that it manages.
+	// +optional
+	GatewayLabels map[string]string `json:"gatewayLabels,omitempty"`
+
+	// RouteAndIngressLabels contains key-value pairs that the ExternalDNS controller will use to filter the HTTPRoutes and Ingresses that it manages.
+	// +optional
+	RouteAndIngressLabels map[string]string `json:"routeAndIngressLabels,omitempty"`
+}
+
+// ExternalDNSConfigurationStatus defines the observed state of ExternalDNSConfiguration.
+type ExternalDNSConfigurationStatus struct {
+	// Conditions is an array of current observed conditions for the ExternalDNSConfiguration
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions"`
+
+	ExternalDNSReadyReplicas       int32 `json:"externalDNSReadyReplicas"`
+	ExternalDNSUnavailableReplicas int32 `json:"externalDNSUnavailableReplicas"`
+
+	// Count of hash collisions for the managed resources. The App Routing Operator uses this field
+	// as a collision avoidance mechanism when it needs to create the name for the managed resources.
+	// +optional
+	// +kubebuilder:validation:Maximum=5
```

Comment:
why a max of 5 here? We aren't providing a configurable name for ExternalDNS so I don't think it makes sense to cap at 5

Reply:
resolving per offline discussion